sdk/src/quotes.ts

```diff
@@ -489,6 +489,180 @@ export const fillQuote = async (
   chainId: ChainId,
   referrer?: string
 ): Promise<Transaction> => {
+  // Use the prepare function to get transaction data
+  const preparedTx = await prepareFillQuote(
+    quote,
+    transactionOptions,
+    wallet,
+    permit,
+    chainId,
+    referrer
+  );
+
+  // Send the transaction
+  const newSwapTx = await wallet.sendTransaction({
+    data: preparedTx.data,
+    to: preparedTx.to,
+    value: preparedTx.value,
+    ...transactionOptions,
+  });
+
+  return newSwapTx;
+};
+
+/**
+ * Function that fills a crosschain swap quote onchain via rainbow's swap aggregator smart contract
+ *
+ * @param {CrosschainQuote} quote
+ * @param {TransactionOptions} transactionOptions
+ * @param {Signer} wallet
+ * @param {string} referrer
+ * @returns {Promise<Transaction>}
+ */
+export const fillCrosschainQuote = async (
+  quote: CrosschainQuote,
+  transactionOptions: TransactionOptions,
+  wallet: Signer,
+  referrer?: string
+): Promise<Transaction> => {
+  // Use the prepare function to get transaction data
+  const preparedTx = await prepareFillCrosschainQuote(quote, referrer);
+
+  // Send the transaction
+  const swapTx = await wallet.sendTransaction({
+    data: preparedTx.data,
+    to: preparedTx.to,
+    value: preparedTx.value,
+    ...transactionOptions,
+  });
+
+  return swapTx;
+};
+
+export const getQuoteExecutionDetails = (
+  quote: Quote,
+  transactionOptions: TransactionOptions,
+  provider: StaticJsonRpcProvider
+): QuoteExecutionDetails => {
+  const instance = new Contract(
+    getRainbowRouterContractAddress(quote.chainId),
+    RainbowRouterABI,
+    provider
+  );
+
+  const {
+    sellTokenAddress,
+    buyTokenAddress,
+    to,
+    data,
+    fee,
+    value,
+    sellAmount,
+    feePercentageBasisPoints,
+  } = quote;
+
+  const ethAddressLowerCase = ETH_ADDRESS.toLowerCase();
+
+  if (sellTokenAddress?.toLowerCase() === ethAddressLowerCase) {
```

Comment:
not sure if this is common practice, I'm used to comparing addresses with an extra utility function like
```typescript
function isEqualHex(hex1: `0x${string}`, hex2: `0x${string}): boolean {
  return BigInt(hex1) === BigInt(hex2)
}
```

This method ensures case insensitivity, and also prevents errors like `0x01 !== 0x1` (like comparing checksum addresses to non-checksum, or is an address constant length in evm?)